src/main/java/io/cdap/plugin/ariba/source/AribaRecordReader.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.ariba.source;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.ariba.source.exception.AribaException;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+
+/**
+ *
+ */
+public class AribaRecordReader extends RecordReader<NullWritable, StructuredRecord> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AribaRecordReader.class);
+  private final AribaServices aribaServices;
+  private final Schema outputSchema;
+  private final List<JsonNode> nodeData = new ArrayList<>();
+  private final AribaStructuredTransformer transformer;
+  private long start;
+  private long end;
+  private long numRowsProcessed;
+  private int pos;
+  private JsonNode row;
+  private ListIterator<JsonNode> jsonNodeListIterator;
+  private JsonNode nodeRecord;
+  private String jobId;
+  private String fileName;
+
+
+  public AribaRecordReader(AribaServices aribaServices, Schema outputSchema) {
+    this.aribaServices = aribaServices;
+    this.outputSchema = outputSchema;
+    this.transformer = new AribaStructuredTransformer();
+  }
+
+  @Override
+  public void initialize(InputSplit split, TaskAttemptContext taContext) {
+    AribaInputSplit aribaInputSplit = (AribaInputSplit) split;
+    LOG.info("Input split file name {} and jobId {}", aribaInputSplit.getFileName(), aribaInputSplit.getJobId());
+    fileName = aribaInputSplit.getFileName();
+    jobId = aribaInputSplit.getJobId();
+  }
+
+  @Override
+  public boolean nextKeyValue() throws IOException {
+    if (jsonNodeListIterator == null) {
+      try {
+        nodeRecord = aribaServices.fetchData(aribaServices.getAccessToken(),
```

Comment:
this code can be moved into initialize()

Reply:
Fixed